ports-of-call/portable_arrays.hpp

```diff
@@ -26,73 +26,142 @@
 
 #include "portability.hpp"
 #include <algorithm>
+#include <array>
 #include <assert.h>
 #include <cstddef> // size_t
 #include <cstring> // memset()
+#include <functional>
+#include <numeric>
+#include <type_traits>
 #include <utility> // swap()
 
+// maximum number of dimensions
+constexpr std::size_t MAXDIM = 6;
+// array type of dimensions/strides
+using narr = std::array<std::size_t, MAXDIM>;
+
+namespace detail {
+
+// set_value end-case
+template <std::size_t Ind, typename NX>
+constexpr void set_value(narr &ndat, NX value) {
+  ndat[Ind] = value;
+}
+
+// set_valueS end case, dispatch to set_value
+template <std::size_t Ind, typename NX>
+constexpr void set_values(narr &ndat, NX value) {
+  set_value<Ind>(ndat, value);
+}
+
+// set_values general case, set `Ind` position of array (tracking from RIGHT)
+// with head of size list `NX`
+template <std::size_t Ind, typename NX, typename... NXs>
+constexpr void set_values(narr &ndat, NX value, NXs... nxs) {
+  set_value<Ind>(ndat, value);
+  set_values<Ind - 1>(ndat, nxs...);
```

Comment:
does this not need a 0 partial specialization to terminate the recursion?

Reply:
The recursion is limited by running out of variadic parameters, or rather `select_value` is the 0 specialization

```c++
set_values(arr, n3, n2,n1) --> set_values(arr, n2, n1) --> set_values(arr, n1) -->set_value(arr, n1)
```

Reply:
Thanks for the explanation @mauneyc-LANL I see it now.